File: 1/1.go
func twoSum(nums []int, target int) []int {

    m := make(map[int]int)

    for i := 0; i < len(nums); i++ {
        temp := int(float64(target - nums[i]))
        if _, ok := m[temp]; ok {
            return []int{m[temp], i}
        } else {
            m[nums[i]] = i
        }
    }

    return nil
}

File: 1/1.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var dicnums = new Dictionary<int, int>();
        for (int i = 0, j; i < nums.Length; i++) {
            if (dicnums.TryGetValue(target - nums[i], out j))
                return new[] {i, j};
            dicnums[nums[i]] = i;
        }
        throw new ArgumentException("No solution found.");
    }
}